staker/stateless_block_validator.go

```diff
@@ -406,30 +381,29 @@ func (v *StatelessBlockValidator) ValidateResult(
 	if err != nil {
 		return false, nil, err
 	}
-	var spawners []validator.ValidationSpawner
-	if useExec {
-		spawners = append(spawners, v.execSpawner)
-	} else {
-		spawners = v.validationSpawners
+	var run validator.ValidationRun
+	if !useExec {
+		if v.redisValidator != nil {
+			if validator.SpawnerSupportsModule(v.redisValidator, moduleRoot) {
+				run = v.redisValidator.Launch(input, moduleRoot)
+			}
+		}
 	}
-	if len(spawners) == 0 {
-		return false, &entry.End, errors.New("no validation defined")
+	if run == nil {
+		for _, spawner := range v.execSpawners {
+			if validator.SpawnerSupportsModule(spawner, moduleRoot) {
+				run = spawner.Launch(input, moduleRoot)
+				break
+			}
+		}
 	}
-	var runs []validator.ValidationRun
-	for _, spawner := range spawners {
-		run := spawner.Launch(input, moduleRoot)
-		runs = append(runs, run)
+	if run == nil {
+		return false, &entry.End, errors.New("this validation not supported by node")
```

Comment:
It seems like a good idea to include the module root in the error here. Also, why return entry.End? It wasn't validated, so I think we should return nil.